### DISCUSSION TITLE
App Mover Script

### DISCUSSION THREAD

**Discussion by holocronology** [007revad/Synology_app_mover#3] [Q&A] [answered]
Q: will this script move the entire docker/container manager to an ssd volume with no additional mucking around?

If yes, then one just needs to move their container files and revise internal mapping to the new volume?

**Comment by 007revad** [comment]
Not yet. 

Container Manager also has an @docker folder on the volume. I will update the script to move @docker as well soon.

**Comment by DrFrankensteinUK** [comment]
This would be a great addition as just getting ready to move my docker setup over to SSD, willing to be the guinea pig..

**Reply by DrFrankensteinUK** [reply]
Actually I can help do testing on my Virtual DSM as need to migrate over the weekend on my main NAS

**Comment by 007revad** [comment]
I spent all yesterday adding and testing support for moving the @docker folder but had a few issues:
1. Occasionally it wouldn't move all the files.
2. Container Manager would list the images but not my containers.
3. When I tried to import my exported container configs it failed because the images didn't actually exist.

To solve issue 1 I changed the script to backup the @docker folder, and used rsync via SSH to copy the @docker folder to the new volume. But I didn't get around to changing the script to use rsync due to issues 2 and 3.

For issue 2 I had to re-download each image which worked for the first two but then all the others failed with a network error.

Also when I tried moving my docker shared folder via control panel it couldn't do it because I have immutable snapshots. I've disabled immutable snapshots and have to wait 7 days before I can delete them.

I'm reluctant to delete all the changes I made, but don't understand docker enough to know what needs to be done.

Thankfully I can more container manager back to volume 1 and restore the @docker backup and container manager works again.

If anyone wants to try or test the changes you can download it here: https://github.com/007revad/Synology_app_mover/blob/test/move_docker_test.sh

**Reply by stevenhaskell** [reply]
@007revad just ran into the issue of trying to move shared folders with immutable snapshots. Is it still the best course of action to disable immutable snapshots, wait 7 days, move the shared folder, then re-enable the immutable snapshots? Thank you in advance!

**Reply by 007revad** [reply]
@stevenhaskell Yes, if the protection period is set to 7 days. if the protection period is set to a different number of days you'd need to wait that long.

**Comment by 007revad** [comment, marked as answer]
I've released a new version moves docker, your image and containers. https://github.com/007revad/Synology_app_mover/releases/tag/v2.0.6

**Comment by brekovcan** [comment]
Line 1560: ERROR Failed to stop Hyper Backup!
This is happening on even Hyper Backup service is running or is already stopped before script.
Using 3.0.31 version of script. DSM 7.1.1 latest.

**Reply by 007revad** [reply]
@brekovcan 

1. Is it only Hyper Backup that gives the error?
2. Are you trying to move, backup or restore Hyper Backup?
3. Do you also have Hyper Backup Vault installed?
4. What does the following command return when Hyper Backup is stopped and when running?
```
sudo -i /usr/syno/bin/synopkg status HyperBackup; echo $?
```

**Reply by brekovcan** [reply]
I have fresh new installed synology. Currently only two packages installed: Hyper Backup and Hyper Backup Vault. I am trying only move from Volume1 to Volume2. Both packages, same error.

Stopping Hyper Backup             
Line 1560: ERROR Failed to stop Hyper Backup!
ash-4.4#  sudo -s ./syno_app_mover.sh
Synology_app_mover v3.0.31
DS713+ DSM 7.1.1-42962-6 

Running from: /volume2/DOWNLOAD/Synology_app_mover-3.0.31/syno_app_mover.sh

1) Move
2) Backup
3) Restore
Select the mode: 1
You selected Move

[Installed package list]
1) /volume1  Hyper Backup
2) /volume1  Hyper Backup Vault
Select the package to move: 2
You selected Hyper Backup Vault in /volume1

Destination volume is /volume2

Ready to Move Hyper Backup Vault to /volume2? [y/n]
y

Stopping Hyper Backup Vault             
Line 1560: ERROR Failed to stop Hyper Backup Vault!
ash-4.4# sudo -i /usr/syno/bin/synopkg status HyperBackup; echo $?
{"aspect":{"active":{"status":"stop","status_description":"translate from systemd status"}},"description":"Status: [0], package is stopped","package":"HyperBackup","status":"stop"}
0

**Reply by 007revad** [reply]
There is an undocumented `--fix` option to bypass the script exiting if a package fails to stop.
```
sudo -s ./syno_app_mover.sh --fix
```

**Reply by 007revad** [reply]
@brekovcan 

I've fixed the "ERROR Failed to stop package" bug for DSM 7.1 and DSM 6.
https://github.com/007revad/Synology_app_mover/releases/tag/v3.0.33

**Comment by holocronology** [comment]
Just complete Container Manager move. Works great. Thanks!
